qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -159,25 +168,118 @@ def num_qubits(self) -> int:
         """Returns the number of qubits used by ansatz and feature map."""
         return self.circuit.num_qubits
 
-    def fit(self, X: np.ndarray, y: np.ndarray):  # pylint: disable=invalid-name
+    def fit(self, X: np.ndarray, y: np.ndarray) -> NeuralNetworkClassifier:
         """
         Fit the model to data matrix X and targets y.
 
         Args:
             X: The input data.
-            y: The target values. Required to be one-hot encoded.
+            y: The target values.
 
         Returns:
-            self: returns a trained classifier.
+            Trained classifier.
         """
-        num_classes = y.shape[-1]
+        y = self._validate_y(y)
+        num_classes = self._get_num_classes(y)
         cast(CircuitQNN, self._neural_network).set_interpret(
             self._get_interpret(num_classes), num_classes
         )
         return super().fit(X, y)
 
+    # pylint: disable=invalid-name
+    def score(self, X: np.ndarray, y: np.ndarray, sample_weight: np.ndarray | None = None) -> float:
+        y = self._validate_y(y)
+        return super().score(X, y, sample_weight=sample_weight)
+
     def _get_interpret(self, num_classes: int):
         def parity(x: int, num_classes: int = num_classes) -> int:
             return x % num_classes
 
         return parity
+
+    def _get_num_classes(self, y: np.ndarray) -> int:
+        """Infers the number of classes from the one-hot encoded targets.
+
+        Args:
+            y: The target values.
+
+        Raises:
+            QiskitMachineLearningError: If the number of classes differs from
+            the previous batch when using a warm start.
+
+        Returns:
+            The number of inferred classes.
+        """
+        if self._one_hot:
+            num_classes = y.shape[-1]
+        else:
+            num_classes = sum(np.unique(y))
+
+        self.num_classes = self._validate_num_classes(num_classes)
+        return self.num_classes
+
+    def _validate_y(self, y: np.ndarray) -> np.ndarray:
+        """Transforms and validates labels.
+
+        Binarizes the labels, infers whether the result is one-hot encoded
+        and sets the relevant attribute in the parent class.
+
+        Args:
+            y: The target values
+
+        Returns:
+            The suitably transformed and validated target values.
+        """
+        targets = self._label_transformer.fit_transform(y)
+        if len(targets[0]) == 1:
+            self._one_hot = False
+            targets = targets.flatten()
+        else:
+            self._one_hot = True
+            self._validate_one_hot_targets(targets)
```

Comment:
So, `self._one_hot` becomes a part of the object state in `VQC` since it is modifiable. While `one_hot` is a read-only parameter in `NeuralNetworkClassifier`. Is it possible to move the label encoding functions from `VQC` to `NeuralNetworkClassifier` then?

Reply:
> One small addition here. Please also consider an additional section in the `NeuralNetworkClassifier/Regressor` tutorial where you could show new features of `VQC`.

Yep, I'm very happy to do that :).

Reply:
Yes, I think so. I'll look into doing that.

Reply:
These changes will now be part of the now-separate PR to fix #320. (This PR's name is now a misnomer.)